src/components/TrialBanner.js

```diff
@@ -150,21 +186,30 @@ export const TrialBanner = _.flow(
             marginLeft: '0.5rem', flexShrink: 0
           },
           onClick: () => {
-            button.isExternal ? window.open(button.url, '_blank') : this.setState({ accessingCredits: true })
+            button.isExternal ? window.open(button.url, '_blank') : this.setState({ openFreeCreditsModal: true })
+          }
+        }, [button.label, button.isExternal ? icon('pop-out', { style: { marginLeft: '0.25rem' } }) : null]),
+        div({
+          style: {
+            marginLeft: '3rem',
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'flex-end'
           }
         }, [
-          loading ? spinner({ style: { fontSize: '1rem', color: 'white' } }) : button.label,
-          button.isExternal ? icon('pop-out', { style: { marginLeft: '0.25rem' } }) : null
-        ]),
-        div({ style: { marginLeft: '3rem', display: 'flex', flexDirection: 'column', alignItems: 'flex-end' } }, [
           h(Clickable, {
             style: { borderBottom: 'none' },
             tooltip: 'Hide banner',
-            onClick: () => this.setState(trialState === 'Terminated' ? { finalizeTrial: true } : { snoozeBanner: true })
+            onClick: trialState === 'Terminated' ? () => this.setState({ finalizeTrial: true }) : () => {
+              localStorage.setItem('removeBanner', 'true')
```

Comment:
In at least some cases, I think it would be good to bring back the banner when the trial state changes. For example, we might want to reintroduce the banner when the trial goes from enrolled to terminated. This could be done by including the trial state in the local storage key.

Probably worth checking with Brad/Ilyana about how it should behave.